unattended-upgrade

```diff
@@ -1119,6 +1182,9 @@ def send_summary_mail(pkgs,                 # type: List[str]
     if (res and apt_pkg.config.find_b(
             "Unattended-Upgrade::MailOnlyOnError", False)):
         return
+    # if the run was successful but nothing had to be done skip sending email
+    if (res and not pkgs and not pkgs_kept_back and not pkgs_removed):
```

Comment:
A test for this would be nice to prevent this from happening again

Reply:
I added the test to autopkgtests.